modules/ding_base/ding_base.module

```diff
@@ -685,3 +685,39 @@ function ding_base_lazy_pane_ajax_deliver($page_callback_result) {
 
   return ajax_deliver($page_callback_result);
 }
+
+/**
+ * Validate an email.
+ *
+ * This check uses Drupals valid_email_addess(), and makes sure the domain used
+ * is a valid top level domain.
+ *
+ * @param String $email
+ *   The email to validate.
+ * @return Boolean
+ *   TRUE if the address is in a valid format.
+ */
+function ding_base_validate_email($email) {
+  if (!valid_email_address($email)) {
+    return FALSE;
+  }
+
+  $tld_list = array();
+  if ($cache = cache_get('ding_base:tld_list')) {
+    $tld_list = $cache->data;
+  }
+  else {
+    // We assume iana will keep the list of tlds in the same format and on the
+    // same URL. It's been the same URL and format since 2004.
+    // If the response fails the $tld_list will be empty and we'll simply skip
+    // the domain check.
+    $response = drupal_http_request('http://data.iana.org/TLD/tlds-alpha-by-domain.txt');
+    if ($response->code == 200) {
+      $tld_list = explode("\n", $response->data);
+      cache_set('ding_base:tld_list', $tld_list);
+    }
+  }
+
+  $domain = strtoupper(substr($email, strrpos($email, '.') + 1));
```

Comment:
If we look at https://tools.ietf.org/html/rfc5322 an valid mail could be "<cableman@test.dk>" with "<>". So as this is placed at a generic checker in ding_base it might take this into account.